prow/github/client.go

```diff
@@ -391,7 +394,13 @@ func (c *Client) doRequest(method, path, accept string, body interface{}) (*http
 	if err != nil {
 		return nil, err
 	}
-	req.Header.Set("Authorization", "Token "+c.token)
+
+	if c.token != nil {
```

Comment:
nil is not allowed

Reply:
I added this check to satisfy the tests.

Reply:
Add the token in the tests that require it vs allowing the separate path here.